app/embedding_server/service.py

```diff
@@ -0,0 +1,39 @@
+import os
+import bentoml
+import numpy as np
+import time
+
+from dotenv import load_dotenv
+from typing import Union
+
+load_dotenv()
+
+
+@bentoml.service
+class SentenceEmbeddingService:
+    def __init__(self) -> None:
+        """초기화 메서드
+        Embedding BentoML 서비스 클래스의 인스턴스를 초기화합니다.
+        """
+        from sentence_transformers import SentenceTransformer
+
+        # bentofile.yaml에 env 변수를 참조하여 특정 모델을 초기화합니다
+        self.embedding_model = SentenceTransformer(
+            os.getenv("EMBEDDING_MODEL"), trust_remote_code=True
+        )
+
+    @bentoml.api
+    def embed(self, sentences: Union[str, list]) -> np.ndarray:
```

Comment:
하나의 API에서 다른 형식(`str`, `list`) 모두를 입력 받도록 구현하면 실사용시 혼동이 있을 수 있습니다.
`list` 형식만 남겨, `["text1", "text2"]` 만 받을 수 있도록 하거나, `str` 형식을 받는 별도 api를 추가하면 어떨까요?

추가적으로 `list`에 대한 타입형 힌트 작성시, `list` element 값의 타입형도 함께 추가해주면 코드이해해 도움이 됩니다. 
```suggestion
    def embed(self, sentences: list[str]) -> np.ndarray:
```

Reply:
[f1241dc](https://github.com/f-lab-edu/llm-mlops/pull/4/commits/f1241dcf35c46508e6cfbee2ac93c3405a0d8501)에 수정했습니다!

Reply:
`list[str]`로 타입형 힌트를 작성했지만 코드상으론 `str`도 입력으로 받을 수 있고, `str` 입력에 대해선 `list[str]` 과는 다른 차원의 아웃풋이 나오지 않나요?

그래서 `list[str]` 타입 입력이 아닌 경우, ValueError등을 raise 해주면 좋을 것 같아요~!

Reply:
[ea860ef](https://github.com/f-lab-edu/llm-mlops/pull/4/commits/ea860efa5d7982fe2a024213d6f311a8a5892365)에서 Pydantic syntax 활용하여 input 형식 체계화했습니다!